core/types/src/macros.rs

```diff
@@ -126,20 +138,54 @@ macro_rules! impl_newtype_for_bytestruct {
             }
         }
 
-        impl ::core::fmt::UpperHex for $wrapper {
-            fn fmt(&self, f: &mut ::core::fmt::Formatter) -> ::core::fmt::Result {
-                let inner: &[u8] = self.as_ref();
-                mc_sgx_util::fmt_hex(inner, f)
+        $crate::derive_debug_and_display_hex_from_as_ref!($wrapper);
+
+    )*}
+}
+
+/// Derive [Debug] and [Display] from [AsRef<T>] to render as a hexadecimal
+/// string.
+///
+/// This is not connected to [ReprBytes] but it is a macro like the above macros
+/// that is often needed for structs holding bytes.
+#[macro_export]
+macro_rules! derive_debug_and_display_hex_from_as_ref {
```

Comment:
This macro was copied from the MobileCoin repo.  The new implementations here broke some tests in the MobileCoin repo due to MrEnclave and MrSigner types not rendering properly. But replacing them seems to have broken some tests here. So the question is: which tests are wrong.